keps/sig-auth/4317-pod-certificates/README.md

```diff
@@ -0,0 +1,1311 @@
+
+**Support private keys shared across multiple pods**: Sharing private keys and certificates across multiple pods may be required in certain applications.  However, this use case is well-served by Secret projected volume sources.
+
+**Support human-in-the-loop approval for certificate requests**: This KEP
+assumes that PodCertificateRequest issuance is fully-automated, because this
+process blocks pod startup for pods that use PodCertificate projected volume
+sources.  PodCertificateRequests do not separate the approval and issuance
+steps, unlike CertificateSigningRequests.
+
+**Standardize on using the SPIFFE SVID format in core Kubernetes**: The certificates issued by `kubernetes.io/kube-apiserver-client-pod` can be made compatible with SPIFFE in the future, since they currently don't include any Subject Alternate Name entries.
+
+**Support identity delegation or impersonation**: We do not intend to support
+issuing certificates that workloads can use as intermediate CA certificates in
+order to issue further certificates, nor do we intend to support one privileged
+workload minting certificates on behalf of other workloads.
+
+## Design Details
+
+### PodCertificateRequest Resource
+
+PodCertificateRequest is a new API type in certificates.k8s.io/v1alpha1.  It's a stripped-down version of CertificateSigningRequest that is tailored directly to the pod certificate issuance use case.
+
+PodCertificateRequest is fundamentally a statement that "Pod X is requesting a certificate from Signer Y".  The signer is in complete control of the form of the issued certificate.  As such, PodCertificateRequest objects don't contain a typical X.509 CSR.  Instead, they contain:
+* A public key, PKIX-serialized.  This key must be an RSA key with modulus size
+  3072 or 4096, an ECDSA key with curve P256 or P384, or an ED25519 key.
+* A proof of possession of the corresponding private key.
+  * If the key is an RSA key, then the signature is over the ASCII bytes of the
+	  pod UID, using RSASSA-PKCS1-V1_5-SIGN from RSA PKCS #1 v1.5 (as implemented
+	  by the golang function crypto/rsa.SignPKCS1v15).
+  * If the key is an ECDSA key, then the signature is as described by [SEC 1,
+	  Version 2.0](https://www.secg.org/sec1-v2.pdf) (as implemented by the golang
+	  library function crypto/ecdsa.SignASN1)
+  * If the key is an ED25519 key, then the signature is as described b
```

Comment:
What's "b" here? Is that referring the same as the ECDSA key?

Reply:
Sorry, that's a mistake.  I just fixed it.